packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -34,6 +34,61 @@ import {
 } from './shared/shared-hook-functions';
 import { toSaveSettings } from './to-save-settings';
 
+function mergeHookFunctions(...hookFunctions: IWorkflowExecuteHooks[]): IWorkflowExecuteHooks {
+	const result: IWorkflowExecuteHooks = {
+		nodeExecuteBefore: [],
+		nodeExecuteAfter: [],
+		workflowExecuteBefore: [],
+		workflowExecuteAfter: [],
+		sendResponse: [],
+		nodeFetchedData: [],
+	};
+	for (const hooks of hookFunctions) {
+		for (const key in hooks) {
+			if (hooks.hasOwnProperty(key)) {
+				result[key]!.push(...hooks[key]!);
+			}
```

Comment:
1. Do we need the `hasOwnProperty` check for a key coming from `in`?
2. Can we avoid the assertions? Maybe like this?

```
if (!result[key] || !hooks[key]) continue;
```

Reply:
1. no. type safety should take care of that.
2. sure

Reply:
2. Is this a programmer error? Should we throw instead of continuing?

Reply:
2. this is caused by bad typings on `IWorkflowExecuteHooks`. I'll fix this in the `WorkflowHooks` refactor PR.